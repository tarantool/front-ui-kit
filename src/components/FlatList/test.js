import React from 'react';
import renderer from 'react-test-renderer';
import { css } from '@emotion/css';
import { FlatList, FlatListItem, Text } from '../../index';

describe('FlatList', () => {
  it('renders correctly', () => {
    const styles = {
      row: css`
        display: flex;
        justify-content: space-between;
      `,
      wrap: css`
        padding: 40px;
      `,
    };

    const items = [
      { name: 'John', balance: 200 },
      { name: 'Emily', balance: 1500 },
      { name: 'Michael', balance: 500 },
    ];

    const tree = renderer
      .create(
        <FlatList>
          {items.map((item) => (
            <FlatListItem className={styles.row} key={item.name}>
              <Text>{item.name}</Text>
              <Text variant="h5" tag="span">{`$${item.balance}`}</Text>
            </FlatListItem>
          ))}
        </FlatList>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
