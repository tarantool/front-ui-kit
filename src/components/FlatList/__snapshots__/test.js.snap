// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatList renders correctly 1`] = `
<ul
  className="css-zmcgmm"
>
  <li
    className="css-1pkilt6"
  >
    <span
      className="css-1ggp6x0"
    >
      John
    </span>
    <span
      className="css-1lh30e5"
    >
      $200
    </span>
  </li>
  <li
    className="css-1pkilt6"
  >
    <span
      className="css-1ggp6x0"
    >
      Emily
    </span>
    <span
      className="css-1lh30e5"
    >
      $1500
    </span>
  </li>
  <li
    className="css-1pkilt6"
  >
    <span
      className="css-1ggp6x0"
    >
      Michael
    </span>
    <span
      className="css-1lh30e5"
    >
      $500
    </span>
  </li>
</ul>
`;
